src/org/omegat/core/machinetranslators/MTHUBTranslate.java

```diff
@@ -0,0 +1,412 @@
+            }
+        } catch (Exception e) {
+            System.err.println("IOException: " + e.getLocalizedMessage());
+        }
+    }
+
+    private synchronized List<String> getAvailableLanguageCodes() {
+        if (availableLanguageCodes == null) {
+            setAvailableLanguageCodes();
+        }
+        return availableLanguageCodes;
+    }
+
+    @Override
+    protected String getPreferenceName() {
+       return Preferences.ALLOW_MTHUB_TRANSLATE;
+    }
+
+    @Override
+    public String getName() {
+        return OStrings.getString("MT_ENGINE_MTHUB");
+    }
+
+    @Override
+    public boolean isConfigurable() {
+        return true;
+    }
+
+    @Override
+    public void showConfigurationUI(Window parent) {
+        MTConfigDialog dialog = new MTConfigDialog(parent, getName()) {
+            @Override
+            protected void onConfirm() {
+                String key = panel.valueField1.getText().trim();
+                boolean temporary = panel.temporaryCheckBox.isSelected();
+                setCredential(PROPERTY_API_KEY, key, temporary);
+            }
+        };
+        dialog.panel.valueLabel1.setText(OStrings.getString(
+                                              "MT_ENGINE_MTHUB_API_KEY_LABEL"));
+        dialog.panel.valueField1.setText(getCredential(PROPERTY_API_KEY));
+        dialog.panel.valueLabel2.setVisible(false);
+        dialog.panel.valueField2.setVisible(false);
+        dialog.panel.temporaryCheckBox.setSelected(
+                isCredentialStoredTemporarily(PROPERTY_API_KEY));
+        dialog.show();
+    }
+
+    @Override
+    protected String translate(Language sLang, Language tLang, String text)
+            throws Exception {
+
+        String mthubKey = getCredential(PROPERTY_API_KEY);
+        if (mthubKey == null || mthubKey.isEmpty()) {
+            return OStrings.getString("MT_ENGINE_MTHUB_KEY_NOTFOUND");
+        }
+
+        String prev = getFromCache(sLang, tLang, text);
+        if (prev != null) {
+            return prev;
+        }
+
+        if (this.availableLanguageCodes == null) {
+            this.getAvailableLanguageCodes();
+            if (this.availableLanguageCodes == null) {
+                /*error to obtain the available language,
+                it is a error with the user key*/
+                return OStrings.getString("MT_ENGINE_MTHUB_INVALID_KEY");
+            }
+        }
+
+        String encoding = StandardCharsets.UTF_8.name();
+        try {
+            URLConnection connection = new URL(GT_URL).openConnection();
+            connection.setDoOutput(true);
+            connection.setRequestProperty("Accept-Charset", encoding);
+            connection.setRequestProperty("Content-Type",
+                                        "application/json;charset=" + encoding);
+
+            try (OutputStream output = connection.getOutputStream()) {
+
+                OM.configure(DeserializationFeature.ACCEPT_SINGLE_VALUE_AS_ARRAY,
```

Comment:
If this configuration is always desirable (I assume it is since it's set every time and never unset) then it should be done in a `static` block near where `OM` is initialized.